tide/wall/SVGTiler.cpp

```diff
@@ -39,20 +39,38 @@
 
 #include "SVGTiler.h"
 
+#include "VectorialContent.h"
+
+#include <QThread>
+
 namespace
 {
-const uint tileSize = 512;
+const uint tileSize = 1024;
 }
 
-SVGTiler::SVGTiler( SVGTextureFactoryPtr factory )
-    : LodTiler( factory->getMaxSize(), tileSize )
-    , _factory( std::move( factory ))
+SVGTiler::SVGTiler( SVG& svg )
+    : LodTiler( svg.getSize() * VectorialContent::getMaxScale(), tileSize )
+    , _svg( svg )
 {}
 
 QImage SVGTiler::getCachableTileImage( const uint tileId ) const
 {
     const QRect imageRect = getTileRect( tileId );
     const QRectF zoomRect = getNormalizedTileRect( tileId );
 
-    return _factory->createTexture( imageRect.size(), zoomRect );
+#if TIDE_USE_CAIRO && TIDE_USE_RSVG
```

Comment:
Hmm, I agree that it's a good question. For the PDFs it felt more appropriate to do it here, because none of the backends can render concurrent from multiple threads, and in fact that's also the case for the SVGs. It's just that the GPU backend is very special because we only call it from the Texture upload thread, thus the above "optimized" path. There is also special handling in the SVGSynchronizer to return an SvgGpuImage wrapper for it, which is not super nice too, but I don't see how to do better. The thread-local storage could be moved into the backend, but I am not sure that it is very appropriate either as it may surprise users of the SVG class that calling render() from a different thread implies internally reallocating a new context (a slow operation)..